### ISSUE TITLE
Share Intent Receiver: track.gpx 

### ISSUE THREAD

**Issue by baiti** [cgeo/cgeo#16187]
### Describe your problem!

When I download a track.gpx (as opposed to a waypoints.gpx), cgeo offers itself as an application to open it. If I  do, cgeo hardcodedly treats it as a gpx containing waypoints. It offers me to put them into a list, which obviously is empty in the end.

As far as I understand, it is possible to differentiate gpx data which contains waypoints from those which contains track segments. Please update the implementation such that this differentiation is made and please handle track.gpx such that the track gets loaded as either 'individual route' or 'load gpx from file' ... where I BTW don't really understand the difference but that is another topic.

### How to reproduce?

just download a track.gpx on your Android phone

### Reproducible

Yes

### c:geo Version

2024.09.18

**Comment by fm-sys** [comment]
Duplicate of https://github.com/cgeo/cgeo/issues/14329, https://github.com/cgeo/cgeo/issues/8626, https://github.com/cgeo/cgeo/issues/11722, https://github.com/cgeo/cgeo/issues/10431

Normally I would just close this issue as duplicate, but as there are \*five\* feature requests about such a small feature meanwhile, maybe someone of the team may just take a look at it?

Adding prio "high" cause it seems to be a highly requested feature, hope this is OK.

**Comment by fm-sys** [comment]
Implementation note: "share" as well as "open" intent should be supported.

**Comment by moving-bits** [comment]
@eddiemuc had already started [some refactoring for GPX import](https://github.com/eddiemuc/cgeo/tree/issues/refactor-gpx-15827) (triggered by #15827), but our current GPX import handling is, let's call it "special" :-)

**Comment by eddiemuc** [comment]
> @eddiemuc had already started [some refactoring for GPX import](https://github.com/eddiemuc/cgeo/tree/issues/refactor-gpx-15827) (triggered by #15827), but our current GPX import handling is, let's call it "special" :-)

I have stopped and abandoned this refactoring some weeks ago. Code is too entangled and I don't know how to entangle w/o breaking existing functions (may the be intended or "unintended")

**Comment by moving-bits** [comment]
I guess we would need to create some generalized importer which can parse several types of GPX files in parallel and return a set of Geocaches / Routes / Tracks in one step, and then handle the result set accordingly. Currently those are specialized importers (and the caller only handles what's expected in that context). Not sure if this can be achieved with our current `setStartElementListener()` (etc.) logic.

**Comment by eddiemuc** [comment]
Might be a topic for dev call tomorrow
